commands/diagnose.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+const cli = require('heroku-cli-util')
+const co = require('co')
+const PGDIAGNOSE_HOST = process.env.PGDIAGNOSE_URL || 'https://pgdiagnose.herokai.com'
+
+function * run (context, heroku) {
+  const fetcher = require('../lib/fetcher')(heroku)
+  const host = require('../lib/host')
+  const util = require('../lib/util')
+  const {app, args} = context
+
+  let generateReport = co.wrap(function * (database) {
+    let db = yield fetcher.addon(app, database)
+    db = yield heroku.get(`/addons/${db.name}`)
+    let config = yield heroku.get(`/apps/${app}/config-vars`)
+    let params = {
+      url: config[db.config_vars[0]],
+      plan: db.plan.name,
+      app: db.app.name,
+      database: db.config_vars[0]
+    }
+    if (!util.starterPlan(db)) {
+      params.metrics = yield heroku.get(`/client/v11/databases/${db.name}/metrics`, {host: host(db)})
+    }
+    return yield heroku.post('/reports', {
+      host: PGDIAGNOSE_HOST,
+      body: params
+    })
+  })
+
+  let displayReport = report => {
+    cli.log(`Report ${report.id} for ${report.app}::${report.database}
+available for one month after creation on ${report.created_at}
+`)
+    let display = checks => {
+      if (!checks.length) return
```

Comment:
Is this necessary? If `checks` is empty, the code won't execute the body of the loop and exit, right?